schemaregistry/schemaregistry.md

```diff
@@ -0,0 +1,257 @@
+- Examples:
+  - "This group holds schemas for the fabulous example app."
+
+## 2.2. Schema and Schema Version
+
+Conceptually, a schema is a description of a data structure. Since data
+structures evolve over time, the schema describing them will also evolve over
+time. Therefore, a schema often has multiple versions.
+
+For simple scenarios, the API allows for version management to be automatic and
+transparent. Whenever a schema is updated, a new version number is assigned and
+prior schema versions are retained. The latest available schema is always the
+default version that is retrieved when the API is given just the schema `id`.
+
+A newer schema version might introduce breaking changes or it might only
+introduce careful changes that preserve compatibility. These strategies are not
+subject of this specification, but the API provides a conflict handling
+mechanism that allows an implementation to reject updates that do not comply
+with a compatibility policy, if one has been implemented.
+
+### 2.2.1. Multi-format schema sets
+
+If the schema format has not been restricted at the schema group level, each
+schema may describe the same data structure for encoding in different formats
+concurrently. That implies that the schema managed by this registry is not
+a single document, but MAY be a set of documents that all describe the exact
+same data structure, each in their own particular syntax and using their own 
+type model.
+
+The schema version therefore relates to this abstract notion of schema and the
+described data structure. All documents coexisting within the same version
+SHOULD describe the exact same data structure.
+
+### 2.2.2. Schema attributes
+
+As per the definition above, the schema object is a collection of versions
+of schema documents. It only has an identifier attribute.
+
+#### id
+
+- Type: `String`
+- Description: Identifies the schema.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string
+  - MUST conform with RFC3986/3.3 `segment-nz-nc` syntax
+  - MUST be unique within the scope of the schema registry
+- Examples:
+  - myschema
+  - my-schema
+
+### 2.2.3 Schema version attributes
+
+A schema version is a document. The "body" of a schema version MAY be a text
+document or binary stream. An implementation SHOULD validate whether a
+schema version is valid according to the rules of its format, for instance
+whether it is a valid Avro schema document when the format is Apache Avro.
+
+Within the scope of the schema set, the version is identified by the combination
+of a version number and an optional format identifier. The schema version MAY
+also have an additional, optional unique identifier within the scope of the
+registry.
+
+#### version
+
+- Type: `Integer`
+- Description: The version of the schema. This is a simple counter and tracks
+  the version in the scope of this schema within the schema group. The schema
+  document MAY indicate a schema that follows a different versioning scheme.
```

Comment:
How do we envision this working exactly? This wording feels like we're opening it up for implementations to diverge from the spec without being specific about how they should do that.

Reply:
This means that the schema document itself can use semver or some other "embedded" versioning notion while the API goes strictly by order or when changes have been added.

Reply:
Understood, but wouldn't semver (or other versioning schemes) imply a different data type?

Reply:
I was wondering the same.